contracts/VestingStake.sol

```diff
@@ -158,49 +158,71 @@ contract VestingStake is ReentrancyGuard, Ownable {
     /// @param _amount Number of additional tokens to deposit into the lock
     /// @param _deferUnclaimed If True, leaves any unclaimed vested balance in the staking contract
     function stake(uint256 _duration, uint256 _amount, bool _deferUnclaimed) public nonReentrant {
+        _stake(_duration, _amount, _deferUnclaimed, msg.sender, msg.sender);
+    }
+
+    /// @notice Deposit additional tokens into another users sAXIAL account
+    /// @param _amount Number of additional tokens to deposit into the lock
+    /// @param _userAddr Address of the user to deposit tokens into
+    function grant(uint256 _amount, address _userAddr) public nonReentrant {
+        // Keep track of new user or pre-existing lockout period
+        if (!usersLock.Initialized) {
+            _stake(104 weeks, _amount, true, _userAddr, msg.sender);
```

Comment:
@Auroter My concern here is that people could use this to screw unlocked users by maxing out their lock period with a minuscule amount. Maybe this should only be available if the msg.sender is the owner?

Reply:
just addressed this with  the aquire